components/director/internal/domain/tenant/resolver.go

```diff
@@ -326,7 +328,7 @@ func (r *Resolver) fetchTenant(tx persistence.PersistenceTx, externalID string)
 	if err := tx.Commit(); err != nil {
 		return nil, err
 	}
-	if err := r.fetcher.FetchOnDemand(externalID, ""); err != nil { // will always fail
+	if err := r.fetcher.FetchOnDemand(externalID, ""); err != nil {
```

Comment:
My idea with this PR is to allow the resolver to fetch the tenant from cis if it is not in the DB. What's the point of having this piece of code if it will always fail?
My assumption is that if we resolver invokes the TenantOnDemand API, the handler will check the cis and if the tenant is present, it will create it in the db. If it is not present, then yes, the tenant will be stored without a parent. But is this bad? I also don't see usecase for the lazily loaded tenants - are they real tenants if they are not part in CIS? Or are they fabricated?

The reasoning of this logic is as follows: a stakeholder of ours has created a MAP resource in Atom. Suddenly, the subaccount in the certificate we use to authenticate against compass has disappeared from the DB (haven't found the root cause). If this resolver is working as it is in the PR, then the hydrator would insert the missing subaccount (as it is not in the DB but it is in CIS) using the externalTentantID query that it is making to verify the subaccount from the certificate.